### PR TITLE
feat: use terraform

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

**Review by S-Coyle — APPROVED** [review]
Been testing this today & verified I was able to create and destroy (including destroying nodes created by another user)

**Review by lionel1704 — APPROVED** [review]
Looks great!

File: up.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+node_capacity=${1:-$TWO_MB}
+
+
+terraform apply \
+     -var "do_token=${DO_PAT}" \
+     -var "pvt_key=${1}" \
+     -var "number_of_nodes=${NODE_OF_NODES}" \
+     -var "node_bin=${3}"
+
+aws s3 cp ip-list s3://safe-testnet-tool/ip-list --acl public-read
```
Comment (S-Coyle) [inline up.sh:16]: Noticed that when I cancelled a deploy, i.e. entered `no` at the prompt below:
```
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value:
```

the script still uploads an empty `ip-list` file to S3 and attempts to register hosts, but obviously none exist.

Can these actions be added under `terraform apply`? Or some other way to only execute when the action is performed?

File: README.md [hunk 2]
```diff
@@ -5,11 +5,42 @@ Adults nodes are killed and restarted at random intervals creating network churn
...
+
+Create a network:
+```
+export DO_PAT=<digital-oceaon-personal-access-token>
 export AWS_ACCESS_KEY_ID=<access-key>
 export AWS_SECRET_ACCESS_KEY=<secret-key>
 export AWS_DEFAULT_REGION=<region, e.g. eu-west-2>
-cd scripts/
-./start-testnet
+./up.sh <path-to-your-DO-registered-ssh-key> [number of nodes]
```
Comment (S-Coyle) [inline README.md:21]: Needs a `terraform init` step added here

**Review by S-Coyle — APPROVED** [review]

**Pull request merged**